lib/pluginmanager/install.rb

```diff
@@ -43,9 +43,11 @@ def validate_cli_options!
   # Check if the specified gems contains
   # the logstash `metadata`
   def verify_remote!(gems)
+    Gem.sources = Gem::SourceList.from(gemfile.gemset.sources)
```

Comment:
- first there is no other calls in `lib/pluginmanager/install.rb`, in fact there is **no** calls at all, everything is handled at a lower level.

- I really prefer to explicitly set the sources **where it is needed** instead of globally setting it in an optimistic way hoping it will cover all cases. In the first case it is precise and explicit, you can see the relation between the sources and the call that requires it, in the other case you don't know where it was done before in the code path... there are lots of chances future refactor will break this and hope there is a spec to catch it. 

So, no, I would definitely stick it in the `spec_for_dependency` call.